### ISSUE TITLE
Probelm with new beneficiaries

### ISSUE THREAD

**Issue by Paomare** [moneymanagerex/moneymanagerex#7049]
MMEX version:
 - [ ] 1.7.0
 - [ ] 1.8.0
 - [x] 1.8.1
 - [ ] 1.8.2
 - [ ] 1.9.0
 - [ ] Other (please specify)

Note: bug reporters are expected to have verified the bug still exists
either in the last stable version of MMEX or on updated development code
(master branch).

Operating System:
 - [x] Windows
 - [ x] Mac OSX
 - [ ] Linux 

# Description of the bug
I have this problem with adding new beneficiaries.
I insert a new beneficiary in an operation and I have no problem.
When I reopen the file and find the box empty. If I search for it in the list of beneficiaries I find it but I can't delete it.
If I try to insert it in the previously created line, it doesn't let me insert it giving an error "can not save PAYEE_V1"

I insert a new beneficiary in an operation and I have no problem.
When I reopen the file and find the box empty. If I search for it in the list of beneficiaries I find it but I can't delete it.
If I try to insert it in the previously created line, it doesn't let me insert it giving an error "can not save PAYEE_V1"
The same thing happens to me when I do it on PC and Android.
So now I have movements without beneficiaries and a certain number of beneficiaries that I can no longer use because they cannot be inserted or deleted

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Reproduction steps:

Expected result:

Actual result:

Log test
 PAYEE_V1: Exception constraint failed[2067]: UNIQUE constraint failed: PAYEE_V1.PAYEENAME, {
    "PAYEEID": -1,
    "PAYEENAME": "Segafredo",
    "CATEGID": -1,
    "NUMBER": "",
    "WEBSITE": "",
    "NOTES": "",
    "ACTIVE": 1,
    "PATTERN": ""
}
18:36:34: PAYEE_V1: Exception constraint failed[2067]: UNIQUE constraint failed: PAYEE_V1.PAYEENAME, {
    "PAYEEID": -1,
    "PAYEENAME": "Segafredo",
    "CATEGID": 386,
    "NUMBER": "",
    "WEBSITE": "",
    "NOTES": "",
    "ACTIVE": 1,
    "PATTERN": ""
}

If you have a backtrace for a crash or a warning, paste it here.

**Comment by renato-mmex** [comment]
He who seeks will find: https://github.com/moneymanagerex/moneymanagerex/issues/6992

**Comment by Paomare** [comment]
I read the post, and I had already tried to solve it.
I tried again but I did not get the desired result.
I have many beneficiaries that it does not show me and that are in the list because I entered many new ones when I was on vacation, therefore with the Android app.
I wonder if I have to delete them all including the respective operations and then try to enter them again.
In addition to Android I use a Mac and sometimes a PC
Now I tried to open it on Android and it gave me a very serious database error and it does not open it anymore (version from Google Play 2024,12,22(1054))
I am in great difficulty

**Comment by renato-mmex** [comment]
Please document which MMEX versions you are working with and on which operating systems. This will make troubleshooting / help much easier.
Try the latest Windows beta version: https://ci.appveyor.com/project/whalley/moneymanagerex/builds/51220133/job/po1yx7t6to5n942o/artifacts

**Comment by Paomare** [comment]
Now I'm using version 2024.12.10 for android because the one from google play didn't open the database
For mac I use 1.8.1
The real problem would be to understand how I should proceed to return to a correct starting situation.
I haven't found any other solution in the forum with an explanation of how to proceed

**Comment by renato-mmex** [comment]
You have written that you sometimes use MMEX on a PC. I assume this is a Windows PC, why don't you install the latest beta recommended by me?
The topic was discussed here: https://github.com/moneymanagerex/android-money-manager-ex/issues/1950

**Comment by Paomare** [comment]
On the PC I only use it when I'm at work so I will use it after January 6th.
For android and mac what version can I find and what should I do with the movements that have an empty beneficiary and with the beneficiaries that I find but cannot use?
Thanks

**Comment by renato-mmex** [comment]
If you like, you can send the .mmb to mmexregister@gmail.com and I will try to solve your problem.
Of course I guarantee the confidentiality of the data.

**Comment by n-stein** [comment]
For macOS try the 1.9.0 beta here https://link.ipx.co.uk/mmex-mac-betas

**Comment by renato-mmex** [comment]
@Paomare  Thank you very much for sending me the .mmb
I have tested with the latest Windows 1.9.0 beta. 
I was able to correct the incorrect entries without any problems.
I suspect that you were working with an Android version that caused these incorrect entries.
At the moment it seems to me that it is quite difficult for users to install / use the correct MMEX versions, especially when using the desktop, Android, macOs and also the cloud at the same time.

Here are the incorrect entries in the database:

![image](https://github.com/user-attachments/assets/f272d6fc-d991-45c0-935a-008cd0012b72)

**Comment by luisdev** [comment]
So, for now, is it better to create new databases using MMX Desktop or AMMEX?

**Comment by Paomare** [comment]
Thanks to everyone for the advice
So now I install the beta version for mac suggested by n-stein.
Then after the holidays I will install on PC the version suggested by Renato-mmx
Then I open the .mmb file and correct the movements without beneficiary.

Now which version do you recommend I use for Android?

I will avoid inserting new beneficiaries from Android
Afterwards I have to go back to asd to use the stable versions or I will stay with the betas?

Many thanks and happy holidays

**Comment by guanlisheng** [comment]
if you stay with the v1.9.0 or newer Desktop version, any Android version would be fine (the newer, the better).

**Comment by Paomare** [comment]
Good morning and happy new year
Now I solved it with the beta versions of android and mac.
For the windows version what should I use?
In the future how can I continue to use the updated versions for both mac and pc? Where can I find them?
Thanks

**Comment by whalley** [comment]
See here for how to access latest builds.... https://moneymanagerex.org/docs/downloads/development/

**Comment by Paomare** [comment]
Perfect Thank you very much
I wanted to point out that in the latest beta for PC the translation of the menus into Italian is not complete

**Comment by Paomare** [comment]
Good morning
I wrote in the forum but I don't get solutions
When I operate on a specific account (nexi Simo only) it gives me this
error that I attach.
Whether I make a new movement or if I try to modify one.
I tried to delete and it closed the program creating a log that should have
been sent.
I think this problem prevents me from opening my file on Android which
obviously no longer works
_/)_/)_/)_/)_/)_/)_/)_/)_/)_/)_/)_/)_/)_/)_/)_/)_/)_/)_/)_/)_/)_/)_/)

Paolo Marendon
Via dell'Ancora, 69
48122 Punta Marina Terme (RA)
ITALY


Il giorno gio 9 gen 2025 alle ore 11:57 Mark Whalley <
***@***.***> ha scritto:

**Comment by whalley** [comment]
> Good morning
> 
> I wrote in the forum but I don't get solutions
> 
> When I operate on a specific account (nexi Simo only) it gives me this
> 
> error that I attach.

Attachments sent via mail cannot be seen, you need to upload directly on GitHub. However if this is a different issue from that originally covered by this issue then can I please suggest you open a new issue to track this.
